src/Microsoft.AspNet.OData.Shared/EdmDeltaEntityObjectOfT.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.Contracts;
+using Microsoft.AspNet.OData.Builder;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Represents an <see cref="IEdmChangedObject"/> with no backing CLR <see cref="Type"/>.
+    /// Used to hold the Entry object in the Delta Feed Payload.
+    /// </summary>
+    [NonValidatingParameterBinding]
+    public class EdmDeltaEntityObject<TStructuralType> : EdmDeltaEntityObject, IEdmChangedObject<TStructuralType>
```

Comment:
>EdmDeltaEntityObject<TStructuralType> [](start = 17, length = 37)

why do you still have this class?

Edm* for no-clr-type scenario, we don't have the <TStructuralType>, how can we use it?

If we have thte TStructuralType, we are in typed scenario.